app/services/email.py

```
@@ -0,0 +1,315 @@
+            
+        Raises:
+            ValueError: If validation fails
+        """
+        # Default policy: from == envelope_from (aligned)
+        envelope_from = email_request.envelope_from or email_request.from_address
+        
+        # Validate envelope_from is in allowed list
+        if not self.validate_envelope_from(envelope_from):
+            raise ValueError(
+                f"envelope_from '{envelope_from}' is not in allowed MailFrom list"
+            )
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)